asserts/asserts.go

```diff
@@ -293,6 +315,74 @@ func (at *AtRevision) String() string {
 	return fmt.Sprintf("%s at revision %d", s, at.Revision)
 }
 
+// AtSequence references a sequence forming assertion at a given sequence point,
+// possibly <=0 (meaning not specified) and revision, possibly not known
+// (RevisionNotKnown).
+// Setting Pinned = true means pinning at the given sequence point (which must be
+// set, i.e. > 0). Pinned sequence forming assertion will be updated to the
+// latest revision within same sequence.
+type AtSequence struct {
+	Type        *AssertionType
+	SequenceKey []string
+	Sequence    int
+	Pinned      bool
+	Revision    int
+}
+
+// Unique returns a unique string representing the sequence by its sequence key
+// that can be used as a key in maps.
+func (at *AtSequence) Unique() string {
+	return fmt.Sprintf("%s/%s", at.Type.Name, strings.Join(at.SequenceKey, "/"))
+}
+
+func (at *AtSequence) String() string {
+	var pkStr string
+	if len(at.SequenceKey) != len(at.Type.PrimaryKey)-1 {
+		pkStr = "???"
+	} else {
+		n := 0
+		// omit series if present in the primary key
+		if at.Type.PrimaryKey[0] == "series" {
+			n++
+		}
+		pkStr = strings.Join(at.SequenceKey[n:], "/")
+		if at.Sequence > 0 {
+			sep := "/"
+			if at.Pinned {
+				sep = "="
+			}
+			pkStr = fmt.Sprintf("%s%s%d", pkStr, sep, at.Sequence)
+		} else {
+			pkStr = fmt.Sprintf("%s/?", pkStr)
```

Comment:
I woudln't use the "?",  just leave it out, it should be clear anyway because I expect sequence to be the only numeric part of the key, and there will be no number (at the end) which should be easy to distinguish